src/Language/PureScript/Errors.hs

```diff
@@ -184,6 +185,23 @@ data HintCategory
 
 data ErrorMessage = ErrorMessage [ErrorMessageHint] SimpleErrorMessage deriving (Show)
 
+-- | Get the source span for an error
+errorSpan :: ErrorMessage -> SourceSpan
+errorSpan = fromMaybe (error "errorSpan: no position information") . findHint matchSpan
+  where
+  matchSpan (PositionedError ss) = Just ss
+  matchSpan _ = Nothing
+
+-- | Get the module name for an error
+errorModule :: ErrorMessage -> ModuleName
+errorModule = fromMaybe (error "errorModule: no position information") . findHint matchModule
```

Comment:
Probably not. We should really change the error message type to enforce these few things, but for now, I'll just change this to use `Maybe`.